config/plugins/pbh.py

```diff
@@ -0,0 +1,852 @@
+
+    if (ip_mask is not None):
+        ip_addr_version = ipaddress.ip_address(ip_mask).version
+
+        if (hash_field in hf_v4) and (ip_addr_version != 4):
+            exit_with_error("Error: the value of '--hash-field'='{}' is NOT compatible with the value of '--ip-mask'='{}'".format(hash_field, ip_mask), fg='red')
+
+        if (hash_field in hf_v6) and (ip_addr_version != 6):
+            exit_with_error("Error: the value of '--hash-field'='{}' is NOT compatible with the value of '--ip-mask'='{}'".format(hash_field, ip_mask), fg='red')
+
+
+def ip_mask_hash_field_update_validator(db, hash_field_name, ip_mask, hash_field):
+    """ Function to validate --ip-mask and --hash-field
+        correspondence, during update flow
+
+        Args:
+            db: reference to CONFIG DB,
+            hash_field_name: name of the hash-field,
+            ip_mask: ip address,
+            hash_field: native hash field value
+    """
+
+    if (ip_mask is None) and (hash_field is None):
+        return
+
+    table = db.cfgdb.get_table(pbh_hash_field_tbl_name)
+    hash_field_obj = table[hash_field_name]
+
+    if (ip_mask is None) and (hash_field is not None):
+
+        try:
+            ip_mask = hash_field_obj['ip_mask']
+        except Exception as e:
+            ip_mask = None
+
+        ip_mask_hash_field_correspondence_validator(ip_mask, hash_field)
+
+    if (ip_mask is not None) and (hash_field is None):
+
+        hash_field = hash_field_obj['hash_field']
+
+        ip_mask_hash_field_correspondence_validator(ip_mask, hash_field)
+
+
+def interfaces_list_validator(db, interface_list, is_update):
+    if is_update and (interface_list is None):
+        return
+
+    is_error = False
+    interfaces_splited = interface_list.split(',')
+
+    for intf in interfaces_splited:
+        if intf.startswith('Ethernet'):
+            if not clicommon.is_valid_port(db.cfgdb, intf):
+                is_error = True
+                break
+        elif intf.startswith('PortChannel'):
+            if not clicommon.is_valid_portchannel(db.cfgdb, intf):
+                is_error = True
+                break
+        else:
+            is_error = True
+            break
+
+    if is_error:
+        exit_with_error("Error: invalid value '{}', for '--interface-list' option".format(interface_list), fg="red")
+
+
+@click.group(
+    name='pbh',
+    cls=clicommon.AliasedGroup
+)
+def PBH():
+    """ Configure PBH (Policy based hashing) feature """
+
+    pass
+
+
+@PBH.group(
+    name="hash-field",
+    cls=clicommon.AliasedGroup
+)
+def PBH_HASH_FIELD():
+    """ Configure PBH hash field """
+
+    pass
+
+
+@PBH_HASH_FIELD.command(name="add")
+@click.argument(
+    "hash-field-name",
+    nargs=1,
+    required=True,
+)
+@click.option(
+    "--hash-field",
+    help="Configure native hash field for this hash field",
```

Comment:
@vadymhlushko-mlnx please fix the description according to YANG model

Reply:
@nazariig it is already according to the [YANG model](https://github.com/Azure/sonic-buildimage/pull/7461)

Reply:
@vadymhlushko-mlnx in YANG we have `description "Configures native hash field for this hash field";`